diff -Naur glucose/core/depend.mk minisat/core/depend.mk
--- glucose/core/depend.mk	1970-01-01 01:00:00.000000000 +0100
+++ glucose-syrup-4.1/core/depend.mk	2017-12-12 11:59:30.008000000 +0100
@@ -0,0 +1,24 @@
+Main.o: Main.C Solver.h ../mtl/Vec.h ../mtl/Heap.h ../mtl/Vec.h \
+  ../mtl/Alg.h SolverTypes.h
+Solver.o: Solver.C Solver.h ../mtl/Vec.h ../mtl/Heap.h ../mtl/Vec.h \
+  ../mtl/Alg.h SolverTypes.h ../mtl/Sort.h
+Solver_C.o: Solver_C.C Solver.h ../mtl/Vec.h ../mtl/Heap.h ../mtl/Vec.h \
+  ../mtl/Alg.h SolverTypes.h Solver_C.h
+Main.op: Main.C Solver.h ../mtl/Vec.h ../mtl/Heap.h ../mtl/Vec.h \
+  ../mtl/Alg.h SolverTypes.h
+Solver.op: Solver.C Solver.h ../mtl/Vec.h ../mtl/Heap.h ../mtl/Vec.h \
+  ../mtl/Alg.h SolverTypes.h ../mtl/Sort.h
+Solver_C.op: Solver_C.C Solver.h ../mtl/Vec.h ../mtl/Heap.h ../mtl/Vec.h \
+  ../mtl/Alg.h SolverTypes.h Solver_C.h
+Main.od: Main.C Solver.h ../mtl/Vec.h ../mtl/Heap.h ../mtl/Vec.h \
+  ../mtl/Alg.h SolverTypes.h
+Solver.od: Solver.C Solver.h ../mtl/Vec.h ../mtl/Heap.h ../mtl/Vec.h \
+  ../mtl/Alg.h SolverTypes.h ../mtl/Sort.h
+Solver_C.od: Solver_C.C Solver.h ../mtl/Vec.h ../mtl/Heap.h ../mtl/Vec.h \
+  ../mtl/Alg.h SolverTypes.h Solver_C.h
+Main.or: Main.C Solver.h ../mtl/Vec.h ../mtl/Heap.h ../mtl/Vec.h \
+  ../mtl/Alg.h SolverTypes.h
+Solver.or: Solver.C Solver.h ../mtl/Vec.h ../mtl/Heap.h ../mtl/Vec.h \
+  ../mtl/Alg.h SolverTypes.h ../mtl/Sort.h
+Solver_C.or: Solver_C.C Solver.h ../mtl/Vec.h ../mtl/Heap.h ../mtl/Vec.h \
+  ../mtl/Alg.h SolverTypes.h Solver_C.h
