diff -Naur glucose-syrup-4.1/core/Solver.h glucose-syrup-4.1/core/Solver2.h
--- glucose-syrup-4.1/core/Solver.h	2016-12-08 13:48:26.000000000 +0100
+++ glucose-syrup-4.1/core/Solver2.h	2018-02-10 13:35:30.885281999 +0100
@@ -154,6 +154,10 @@
     void    setPolarity    (Var v, bool b); // Declare which polarity the decision heuristic should use for a variable. Requires mode 'polarity_user'.
     void    setDecisionVar (Var v, bool b); // Declare if a variable should be eligible for selection in the decision heuristic.

+    //NUSMV SEED
+    void    setRandomSeed  (double seed); // Sets random seed (cannot be 0)
+    //NUSMV SEED
+
     // Read state:
     //
     lbool   value      (Var x) const;       // The current value of a variable.
@@ -539,6 +543,7 @@
     decision[v] = b;
     insertVarOrder(v);
 }
+inline void     Solver::setRandomSeed(double seed){assert(seed != 0); random_seed=seed;}
 inline void     Solver::setConfBudget(int64_t x){ conflict_budget    = conflicts    + x; }
 inline void     Solver::setPropBudget(int64_t x){ propagation_budget = propagations + x; }
 inline void     Solver::interrupt(){ asynch_interrupt = true; }
